components/tabs/tabs.js

```diff
@@ -739,6 +792,43 @@ class Tabs extends LocalizeCoreElement(ArrowKeysMixin(SkeletonMixin(LitElement))
 		this.requestUpdate();
 	}
 
+	async _handleTabsSlotChange(e) {
+		this._defaultSlotBehavior = false;
+
+		this._tabs = e.target.assignedNodes({ flatten: true })
+			.filter((node) => node.nodeType === Node.ELEMENT_NODE && node.role === 'tab');
+
+		// handle case where there are less than two tabs initially
+		this._updateTabListVisibility(this._tabs);
+
+		if (!this._initialized && this._tabs.length === 0) return;
+
+		let selectedTab;
+		if (this._tabs.length > 0) {
+			selectedTab = this._tabs.find((tab) => tab.state !== 'removing');
+			if (selectedTab) {
+				selectedTab.selected = true;
+				selectedTab.tabIndex = 0;
+			}
+		}
+
+		await this.updateComplete;
+
+		if (!this._initialized && this._tabs.length > 0) {
+			this._initialized = true;
+		}
+
+		if (selectedTab) {
+			// set corresponding panel to selected
+			const selectedPanel = this._getPanel(selectedTab.id);
+			if (selectedPanel) selectedPanel.selected = true;
+		}
+
+		this.dispatchEvent(new CustomEvent(
+			'd2l-tabs-initialized', { bubbles: true, composed: true }
```

Comment:
Is this event used? Not really wanting to introduce extra risk, but it's not coming up in any searches.

Reply:
Doesn't look to be! I'll clean that up